tests/evaluate/test_default_evaluator.py

```diff
@@ -3022,3 +3022,28 @@ def test_evaluate_with_correctness():
                 "correctness/v1/variance": 0.0,
                 "correctness/v1/p90": 3.0,
             }
+
+
+def test_evaluate_custom_metrics_string_values():
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"text": ["Hello world", "My name is MLflow"]})
+        results = mlflow.evaluate(
+            model_info.model_uri,
+            data,
+            extra_metrics=[
+                make_metric(
+                    eval_fn=lambda predictions, metrics, eval_config: MetricValue(
+                        aggregate_results={"eval_config_value_average": eval_config}
+                    ),
+                    name="cm",
+                    greater_is_better=True,
+                    long_name="custom_metric",
+                )
+            ],
+            evaluators="default",
+            evaluator_config={"eval_config": 3},
+        )
+        assert results.metrics["cm/eval_config_value_average"] == 3
```

Comment:
would it be possible to add to the existing tests for `_recall`, `_precision`, and `_f1_score` so that the `average` parameter with a different string value is also tested?

Reply:
NVM on this since the classifier and regressor uses a different codepath that doesn't use `_get_args_for_metrics`

Reply:
I think the problem is all of the above metrics have defaults. So it is very difficult to test this usecase to see if the metrics was called with defaults or with values passed from mlflow.evaluate without actually stubbing the metrics.

So I thought instead of stubbing it. I can just create a custom metric and test that instead.